docs/select-expression.md

```diff
@@ -178,14 +179,16 @@ suspend fun selectAorB(a: ReceiveChannel<String>, b: ReceiveChannel<String>): St
             else 
                 "a -> '$value'"
         }
-        b.onReceiveOrNull { value -> 
-            if (value == null) 
+        b.onReceiveOrClosed { valueOrClosed -> 
+            if (valueOrClosed.isClosed) 
                 "Channel 'b' is closed"
             else    
-                "b -> '$value'"
+                "b -> '${valueOrClosed.value}'"
         }
     }
 ```
+Note that [onReceiveOrNull][ReceiveChannel.onReceiveOrNull] is an extension function defined only for channels with non-nullable elements.
+
```

Comment:
This seems to be a good place in documentation to add an extended rationale on _why_ it is defined to be so (so that generic functions that are designed to operation on any `T` do not accidentally confuse closed channel and the null element)